Content.Server/Chemistry/Components/MeleeChemicalPickerComponent.cs

```diff
@@ -0,0 +1,25 @@
+using Content.Shared.FixedPoint;
+
+namespace Content.Server.Chemistry.Components
+{
+    [RegisterComponent]
+    public sealed partial class MeleeChemicalPickerComponent : Component
+    {
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("transferAmount")]
+        public FixedPoint2 TransferAmount { get; set; } = FixedPoint2.New(1);
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float TransferEfficiency { get => _transferEfficiency; set => _transferEfficiency = Math.Clamp(value, 0, 1); }
+
+        [DataField("transferEfficiency")]
+        private float _transferEfficiency = 1f;
+
+        /// <summary>
+        ///     Solution to inject to.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("solution")]
```

Comment:
datafields infer the name if none is specified, so change all of the ones in this component from:
`[DataField("name"]` to `[DataField]`